core/src/main/java/apoc/meta/Meta.java

```diff
@@ -909,39 +909,59 @@ private void filterNonExistingRelationships(Map<Pattern, Relationship> vRels, Me
                 .forEach(vRels::remove);
     }
 
-    private boolean relationshipExists(Pattern p, Relationship relationship, MetaConfig metaConfig) {
-        if (relationship==null) return false;
-        double degreeFrom = (double)(long)relationship.getProperty("out")  / (long)relationship.getStartNode().getProperty("count");
-        double degreeTo = (double)(long)relationship.getProperty("in")  / (long)relationship.getEndNode().getProperty("count");
+    private boolean relationshipExists(Pattern p, Relationship relationship, SampleMetaConfig metaConfig) {
+        if (relationship == null) return false;
+        double degreeFrom = (double)(long)relationship.getProperty("out")  / (long) relationship.getStartNode().getProperty("count");
+        double degreeTo = (double)(long)relationship.getProperty("in")  / (long) relationship.getEndNode().getProperty("count");
 
+        Map<String, Long> countStore = getLabelCountStore(tx, kernelTx);
         if (degreeFrom < degreeTo) {
-            if (relationshipExists(p.labelFrom(), p.labelTo(), p.relationshipType(), Direction.OUTGOING, metaConfig)) return true;
+            return relationshipExists(tx, countStore, p.labelFrom(), p.labelTo(), p.relationshipType(), Direction.OUTGOING, metaConfig);
         } else {
-            if (relationshipExists(p.labelTo(), p.labelFrom(), p.relationshipType(), Direction.INCOMING, metaConfig)) return true;
+            return relationshipExists(tx, countStore, p.labelTo(), p.labelFrom(), p.relationshipType(), Direction.INCOMING, metaConfig);
         }
-        return false;
     }
 
-    private boolean relationshipExists(Label labelFromLabel, Label labelToLabel, RelationshipType relationshipType, Direction direction, MetaConfig metaConfig) {
-        Map<String, Long> countStore = getLabelCountStore();
+    /**
+     * relationshipExists uses sampling to check if the relationships added in previous steps exist.
+     * The sample count is the skip count; e.g. if set to 1000 this means every 1000th node will be checked.
+     * A sample count higher than nodeLabel count is equivalent to not checking them.
+     * Note; Each node is still fetched, but the relationships on that node will not be checked
+     * if skipped, which should make it faster.
+     */
+    static boolean relationshipExists(
```

Comment:
I wonder if it would be easier to understand if not both this and the previous method had the same name?

Reply:
Most likely! will update